run

```diff
@@ -234,6 +234,30 @@ retry() {
     return 0
 }
 
+get_arch() {
+    local platform="${1:-$(uname -p)}"
```

Comment:
`uname -m` may be more portable?
I think it will mostly be `x86_64` or `aarch64`, but it looks like there are other arm-related possibilities listed here: https://stackoverflow.com/questions/45125516/possible-values-for-uname-m